test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1265,6 +1390,30 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
         ASSERT_UCS_OK(ucp_worker_set_am_recv_handler(e.worker(), &param));
     }
 
+protected:
+    enum {
+        SEND_STOP = UCS_BIT(0),
+        RECV_STOP = UCS_BIT(1)
+    };
+
+    static void recv_disconnect(void *arg) {
+        test_ucp_sockaddr_protocols *test =
+            static_cast<test_ucp_sockaddr_protocols*>(arg);
+        test->one_sided_disconnect(test->receiver(), UCP_EP_CLOSE_MODE_FORCE);
+        while (m_err_count == 0) {
+            test->short_progress_loop();
+        }
+    }
+
+    static void send_disconnect(void *arg) {
+        test_ucp_sockaddr_protocols *test =
+            static_cast<test_ucp_sockaddr_protocols*>(arg);
+        test->one_sided_disconnect(test->sender(), UCP_EP_CLOSE_MODE_FORCE);
+        while (m_err_count == 0) {
+            test->short_progress_loop();
+        }
```

Comment:
seems code duplication
instead of passing stop_cb_t , maybe pass entity* to disconnect?